MapboxMobileEvents/MMEEvent.m

```diff
@@ -135,6 +134,27 @@ + (instancetype)crashEventReporting:(NSError *)eventsError error:(NSError **)cre
             index++;
         }
         crashAttributes[MMEEventKeyStackTrace] = callStack;
+
+        /* compute a hash of the full trace */
+        NSData *callstackDigest = [[errorException.callStackSymbols componentsJoinedByString:@"+"] dataUsingEncoding:NSUTF8StringEncoding];
+        if (callstackDigest) {
+            uint8_t digest[CC_SHA224_DIGEST_LENGTH];
+            CC_SHA224(callstackDigest.bytes, (unsigned)callstackDigest.length, digest);
+            NSMutableString* output = [NSMutableString stringWithCapacity:CC_SHA224_BLOCK_BYTES];
+            for(int i = 0; i < CC_SHA224_DIGEST_LENGTH; i++) {
+                [output appendFormat:@"%02x", digest[i]];
+            }
+
+            if (output) {
+                crashAttributes[MMEEventKeyStackTraceHash] = output;
```

Comment:
What's the hash used for? Is it just to "bucket" crashes together?

Reply:
Because we redact out any stack frames that don't come from our own libs or a couple of the system libs, we can't necessarily distinguish between two traces based on the stack trace we get back.

This hash is computed across the entire trace, including symbols, so that we can isolate how many crashes are happening in the same build of a particular app. The android SDK does something similar so this was added for parity.